website/registrations/utils.py

```diff
@@ -0,0 +1,349 @@
+                                             validations=self.validations,
+                                             log_error=functools.partial(log_error, header=self.header))
+
+    def validate(self):
+        self.field.parse()
+
+    def get_parsed_value(self):
+        return {self.header: self.field.parse()}
+
+    def get_raw_value(self):
+        return self.value
+
+    def to_json(self):
+        return json.dumps({self.header: self.field.parse()})
+
+    @classmethod
+    def field_instance_for(cls, name, **kwargs):
+        field_instance = None
+        if name in METADATA_FIELDS.keys():
+            if name in CONTRIBUTOR_METADATA_FIELDS:
+                field_instance = ContributorField(**kwargs)
+            elif name == 'license':
+                field_instance = LicenseField(**kwargs)
+            elif name == 'category':
+                field_instance = CategoryField(**kwargs)
+            elif name == 'subjects':
+                field_instance = SubjectsField(**kwargs)
+            elif name == 'affiliated institutions':
+                field_instance = InstitutionsField(**kwargs)
+            elif name == 'project guid':
+                field_instance = ProjectIDField(**kwargs)
+            else:
+                field_instance = MetadataField(**kwargs)
+        else:
+            field_instance = RegistrationResponseField(**kwargs)
+        return field_instance
+
+class RegistrationResponseField():
+    def __init__(self, **kwargs):
+        response_validations = kwargs['validations']
+        self.required = response_validations.get('required', False)
+        self.type = response_validations.get('type', 'string')
+        self.format = response_validations.get('format')
+        self.options = response_validations.get('options', [])
+        self.value = kwargs.get('value', '').strip()
+        self.log_error = kwargs['log_error']
+        self._parsed_value = None
+
+    def _validate(self):
+        parsed_value = None
+        if self.required and not bool(self.value):
+            self.log_error(missing=True)
+        else:
+            if self.type == 'string':
+                parsed_value = self.value
+            elif self.type == 'choose' and self.format in ['singleselect', 'multiselect']:
+                if self.format == 'singleselect':
+                    if self.value not in self.options:
+                        self.log_error(invalid=True)
+                    else:
+                        parsed_value = self.value
+                else:
+                    parsed_value = []
+                    choices = [val.strip() for val in self.value.split(';')]
+                    for choice in choices:
+                        if choice not in self.options:
+                            self.log_error(invalid=True)
+                        else:
+                            parsed_value.append(choice)
+            self._parsed_value = parsed_value
+
+    def parse(self):
```

Comment:
There could be a `Field` class from which both `RegistrationResponseField` and `MetadataField` inherits from, so that they have this `parse` method for free.
Also I'd recommend changing the name to `validate_and_parse`, because this method actually do both.